packages/repository/test/acceptance/belongs-to.relation.acceptance.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {juggler, repository, RepositoryMixin, AppWithRepository} from '../..';
+import {CustomerRepository, OrderRepository} from '../fixtures/repositories';
+import {expect} from '@loopback/testlab';
+import {Application} from '@loopback/core';
+
+describe('HasMany relation', () => {
+  // Given a Customer and Order models - see definitions at the bottom
+
+  let app: AppWithRepository;
+  let controller: OrderController;
+  let customerRepo: CustomerRepository;
+  let orderRepo: OrderRepository;
+
+  before(givenApplicationWithMemoryDB);
+  before(givenBoundCrudRepositoriesForCustomerAndOrder);
+  before(givenOrderController);
+
+  afterEach(async () => {
+    await orderRepo.deleteAll();
+  });
+
+  it('can find customer of order', async () => {
+    const customer = await customerRepo.create({name: 'Order McForder'});
+    const order = await orderRepo.create({
+      customerId: customer.id,
+      description: 'Order from Order McForder, the hoarder of Mordor',
+    });
+    const result = await controller.findOwnerOfOrder(order.id);
+    expect(result).to.deepEqual(customer);
+  });
+
+  //--- HELPERS ---//
+
+  class OrderController {
+    constructor(
+      @repository(OrderRepository) protected orderRepository: OrderRepository,
+    ) {}
+
+    async findOwnerOfOrder(orderId: number) {
+      const order = await this.orderRepository.findById(orderId);
+      return await this.orderRepository.customer(order);
```

Comment:
See my earlier comment.

This implementation requires two database calls.

Ideally, `findOwnerOfOrder` should make a single database query only. 

```sql
# a mock up, may not work
SELECT * 
  FROM Customer 
  JOIN Order ON Order.customerId = Customer.id
  WHERE Order.id = ${orderId}
```

Even if our implementation cannot do such SQL JOIN query yet, the API should be designed in a way that will allow such implementation in the future.

For example:

```ts
// a mock-up implementation of belongsTo "find" function
// assuming we can query related properties
const data = await this.customerRepository.find({
  include: ['orders'],
  where: {'order.id': orderId},
  fields: {orders: true}
});
return data.orders[0];
```